basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -0,0 +1,880 @@
+                {
+                    throw std::invalid_argument("lane following tactical plugin doesn't support this maneuver type");
+                }
+                else if(std::find(lane_change_plugins_list.begin(), lane_change_plugins_list.end(), general_config.trajectory_type) !=lane_change_plugins_list.end() && maneuver.type != cav_msgs::Maneuver::LANE_CHANGE)
+                {
+                    throw std::invalid_argument("lane change tactical plugin doesn't support this maneuver type");
+                }
+                else{
+                    double starting_downtrack = GET_MANEUVER_PROPERTY(maneuver, start_dist);
+                    if(first){
+                        starting_downtrack = std::min(starting_downtrack, max_starting_downtrack);
+                        first = false;
+                    }
+                    ROS_DEBUG_STREAM("Used downtrack: " << starting_downtrack);
+
+                    if(maneuver.type == cav_msgs::Maneuver::LANE_FOLLOWING){
+                        ROS_DEBUG_STREAM("Creating Lane Follow Geometry");
+                        std::vector<PointSpeedPair> lane_follow_points = create_lanefollow_geometry(maneuver, max_starting_downtrack, wm, ending_state_before_buffer, general_config, detailed_config);
+                        points_and_target_speeds.insert(points_and_target_speeds.end(), lane_follow_points.begin(), lane_follow_points.end());
+                    }
+                    else if(maneuver.type == cav_msgs::Maneuver::LANE_CHANGE){
+                        ROS_DEBUG_STREAM("Creating Lane Change Geometry");
+                        std::vector<PointSpeedPair> lane_change_points = create_lanechange_geometry(maneuver, max_starting_downtrack, wm, ending_state_before_buffer, state, detailed_config);
+                        points_and_target_speeds.insert(points_and_target_speeds.end(), lane_change_points.begin(), lane_change_points.end());
+                    }
+                    else{
+                        throw std::invalid_argument("This maneuver type is not supported");
+                    }
+                }
+            }
+
+            //Add buffer ending to lane follow points at the end of maneuver(s) end dist 
+            if(maneuvers.back().type == cav_msgs::Maneuver::LANE_FOLLOWING){
+                points_and_target_speeds = add_lanefollow_buffer(wm, points_and_target_speeds, maneuvers, ending_state_before_buffer, detailed_config);
+
+            }
+            return points_and_target_speeds;
+
+        }
+
+        std::vector<PointSpeedPair> create_lanefollow_geometry(const cav_msgs::Maneuver &maneuver, double starting_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm, cav_msgs::VehicleState &ending_state_before_buffer,
+                                                                    const GeneralTrajConfig &general_config, const DetailedTrajConfig &detailed_config)
+        {
```

Comment:
Might not hurt to defensively check here that the config's maneuver type matches lane follow, just to be sure.

Reply:
Would this be the same check as line 33 and line 37?